domino-ui/src/main/java/org/dominokit/domino/ui/upload/FileItem.java

```diff
@@ -352,7 +352,9 @@ private void onError() {
     private String getErrorMessage() {
         if (errorMessage != null)
             return errorMessage;
-        return request.responseText.isEmpty() ? "Error while sending request" : request.responseText;
+        
+        final boolean hasErrorText = request.responseType != null && (request.responseType.isEmpty() || request.responseType.equals("text")) && !request.responseText.isEmpty();
+        return hasErrorText ? request.responseText : "Error while sending request";
```

Comment:
We only need to return the error message if it is not null and a default message if null.